src/stream.rs

```diff
@@ -435,13 +435,21 @@ pub(crate) fn connect_host(
 
             let mut proxy_response = Vec::new();
 
-            loop {
-                let mut buf = vec![0; 256];
-                let total = stream.read(&mut buf)?;
-                proxy_response.append(&mut buf);
-                if total < 256 {
+            let mut buf = [0u8; 1];
+            while let Ok(total) = stream.read(&mut buf) {
```

Comment:
The `?` on `read` in previous version propagated the IO error out. This stops the loop on `Err`, silently discards it, which is probably not what we want. I suggest keep using the `loop + read` variant.